### ISSUE TITLE
Proposed Selector Augments (Allows Automatic Switching)

### ISSUE THREAD

**Issue by WebSoak** [TeamOpenIndustry/ImmersiveRailroading#1248]
-=[Describe the suggestion]=-
Hello, I have come across the issue of switches being ignorant of the train's destination. I would like to propose a fix for this issue. This was a concept I had solved with minecarts many months ago, and the concept could be applied to the Immersive Railroading detector augment system as well. Currently, there is no way for the switching process to be done without stopping the train, exiting the train, pressing a button on a switch control board, then re-entering the train. It is quite un-realistic for the driver of a train to exit the train in order to change the switch, especially in instances where the switch might be located far between stations. However, this could be fixed via the addition of a one to two digit based identification system applied to a new selector augment & the trains. How would this work? The selector augment would have a GUI that can opened, giving players the option to input a 1 to 2 digit code. The trains would be outfitted with the same GUI option. If the codes match, then the selector augment will output a signal. Meaning, a player could hook the selector augment up to the switch via red stone and trains will automatically be switched onto the appropriate track and follow the player's desired route. All the train operator would need to do is know the correct ID code for the selector augment and be on their way. No exiting of the train required, and the player wouldn't even need to set up switch boards or levers to control their switches; because the switches could be easily automated.
This very short 3 minute video explains how this concept was already applied to minecarts and was used to create automated switches in the past. The concept would apply the same to IR tracks: https://www.youtube.com/watch?v=fvJuaSXUyFQ 
 
-=[Applications of the suggestion]=- 
Simply put, this would allow for automated switching as well as route identification of the trains by selector augments without the requirement of assigned rolling stock, which gives players more freedom with making their switches amongst other creative uses for this augment.

![2022-06-13_19 09 39](https://user-images.githubusercontent.com/79772407/173462138-cc726427-712b-4399-917f-3e9dc53e3745.png)
![2022-06-13_19 09 48](https://user-images.githubusercontent.com/79772407/173462145-4bc411cc-2da5-4338-a6e7-b935ee55ad3c.png)
![2022-06-13_19 09 58](https://user-images.githubusercontent.com/79772407/173462156-d08aa576-d19a-42c0-a642-42abceae69ed.png)
![2022-06-13_19 10 38](https://user-images.githubusercontent.com/79772407/173462182-69b8185a-f391-4129-aa20-06cfa2807b56.png)

**Comment by cam72cam** [comment]
This can now be done with labels and augments
